Comments.md

```diff
@@ -0,0 +1,148 @@
+public enum ShimmerStyle {
+    case concealing
+    case revealing
+}
+
+```
+
+### Good: clear documentation that explains the different enum cases
+
+``` swift
+/// Shimmer style can be either concealing or revealing
+/// The style affects how shimmer is animated 
+public enum ShimmerStyle {
+    /// Concealing shimmer: the gradient conceals parts of the subviews as it moves leaving most parts of the subviews unblocked.
+    case concealing
+
+    /// Revealing shimmer: the gradient reveals parts of the subviews as it moves leaving most parts of the subview blocked.
+    case revealing
+}
+```
+
+## Convention - Inline Comments
+There are three main purposes of inline comments:
+- Explain why the code is implemented in a certain way and give extra context.
+- Explain complex code that's not obvious.
+- Break up code blocks into chunks (when refactoring is not desirable).
+
+## Rationale
+Adding inline comments brings clarity and readability to complex code and logic.
+
+## Examples
+
+### Bad: does not explain why the the expand() function is called under a specific condition
+
+``` swift
+class SomeView {
+    ...
+    override func mouseEntered(with event: NSEvent) {
+        if !animationPlaying {
+            expand()
+        }
+    }
+
+    func expand() {
+        ...
+    }
+}
+```
+
+### Good: explains why the expand() function is called under a specific condition
+
+``` swift
+class SomeView {
+    ...
+    override func mouseEntered(with event: NSEvent) {
+        // There are multiple mouse enter events fired during the expand/collapse animation because the tracking area is constantly changing. Only respond to the initial event.
+        // Only respond to the events when animation is not playing
+        if !animationPlaying {
+            expand()
+        }
+    }
+
+    func expand() {
+        ...
+    }
+}
+```
+
+### Bad: no explanation on how the finalScore is calculated
+
+``` swift
+let finalScore = scores.reduce(0, combine: +) / Double(scores.count)
+}
+```
+
+### Good: explain in an easily understandable way how finalScore is calculated
+
+``` swift
+// Compute the final score as the arithmetic mean of the previous scores
+let finalScore = scores.reduce(0, combine: +) / Double(scores.count)
+}
+```
+
+### Bad: one big chunk of code
+
+``` swift
+func loadLastCheckpoint() -> GameState? {
+    let defaultDataPath = getDefaultDataPath()
+    guard let saveFilePath = defaultDataPath.appendingPathComponent("SaveFile.dat") else {
+        return nil
+    }
+    guard let dataContents = try? NSData(contentsOfFile: saveFilePath) else {
+        return nil
+    }
+    guard let loadedState = GameState.deserialize(dataContents) else {
+        return nil
+    }
+    loadedState.resetToLastCheckpoint()
+    return loadedState
+}
+```
+
+### Good: break up code blocks into chunks
+
+``` swift
+func loadLastCheckpoint() -> GameState? {
+    // Determine the path to the file on disk
+    let defaultDataPath = getDefaultDataPath()
```

Comment:
Convention is still to avoid "get*" getters right? I know it's not what this guideline is about, but all code in the guide should adhere to all style guidelines we have and inherit